jobs/distributable_lib.tcl

```diff
@@ -219,26 +220,34 @@ proc check_licenses {portName variantInfo} {
         set sub_names {}
         foreach full_lic $sublist {
             # chop off any trailing version number
-            set lic [remove_version [string tolower $full_lic]]
+            set lic [remove_version $full_lic]
             # add name to the list for later
             lappend sub_names $lic
-            if {[info exists ::license_good($lic)]} {
+            if {[info exists ::license_good([string tolower $lic])]} {
                 set any_good 1
             }
         }
         lappend top_license_names $sub_names
         if {!$any_good} {
-            return [list 1 "\"$portName\" is not distributable because its license \"$lic\" is not known to be distributable"]
+            lappend failures "\"$portName\" is not distributable because its license \"$lic\" is not known to be distributable"
         }
     }
 
     # start with deps of top-level port
+    set portPaths [dict create [lindex $top_info 0] [list]]
     set portList [lindex $top_info 0]
+    foreach aPort $portList {
+        dict set portPaths $aPort [list]
+    }
+
     while {[llength $portList] > 0} {
         set aPort [lindex $portList 0]
+        set portList [lreplace $portList 0 0]
+        if {[info exists portSeen($aPort)] && $portSeen($aPort) eq 1} {
```

Comment:
`eq` is for string comparisons and strings should always be quoted in expressions. However there's really no need to check the value at all.